### PR TITLE
fix(tracing): properly resolve when multiple trace agent url env variables are used

### LINKED ISSUE #3648: Datadog k8s Admission Controller makes using Unix Domain Socket impossible
### Which version of dd-trace-py are you using?

1.0.2

### Which version of pip are you using?

21.3.1

### Which version of the libraries are you using?

datadog==0.44.0
ddtrace==1.0.2

### How can we reproduce your problem?

[Enable the Admission Controller](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml#L694) in the Datadog helm chart: 
```
clusterAgent.admissionController.enabled=true
```

Start up a container with `ddtrace` installed:
```
# kubectl run ... -- bash
```

The Datadog Admission Controller automagically sets various Datadog `DD_`* environment variables on the container.
```
# env | grep DD_
DD_AGENT_HOST=10.103.1.28
DD_ENTITY_ID=3aabef08-143d-4396-9b51-9811354f6adf
```

Try and start python with ddtrace:
```
# DD_TRACE_STARTUP_LOGS=true ddtrace-run python
- DATADOG TRACER DIAGNOSTIC - Agent not reachable at http://10.103.1.28:8126. Exception raised: [Errno 111] Connection refused
Python 3.8.3 (default, May 19 2020, 18:47:26)
...
```

Set `DD_TRACE_AGENT_URL` to explicitly use UDS:
```
# DD_TRACE_STARTUP_LOGS=true DD_TRACE_AGENT_URL=unix:///run/datadog/apm.socket  ddtrace-run python
- DATADOG TRACER DIAGNOSTIC - Agent not reachable at http://10.103.1.28:None. Exception raised: Port could not be cast to integer value as 'None'
Python 3.8.3 (default, May 19 2020, 18:47:26)
...
```

Note how the error has changed, so setting `DD_TRACE_AGENT_URL` has an effect but does not switch ddtrace to using UDS. 

Remove the `DD_AGENT_HOST` env var and try again:
```
# unset DD_AGENT_HOST
# DD_TRACE_STARTUP_LOGS=true DD_TRACE_AGENT_URL=unix:///run/datadog/apm.socket  ddtrace-run python
Python 3.8.3 (default, May 19 2020, 18:47:26)
...
```

Now it works. And setting `DD_TRACE_AGENT_URL` is not required:
```
# DD_TRACE_STARTUP_LOGS=true  ddtrace-run python
Python 3.8.3 (default, May 19 2020, 18:47:26)
...
```

I want the other environment variables the Admission Controller sets up, but not `DD_AGENT_HOST`. I thought I could achieve this and work around the issue by setting the `DD_AGENT_HOST` to empty string in the pod manifest. But surprisingly that still did not work. This still produces `Agent not reachable at http://10.103.1.28:8126. Exception raised: [Errno 111] Connection refused`. 

If I disable the Datadog Admission Controller then it works. But obviously I have to manually reproduce all the other env vars it would add for me.

### What is the result that you expected?

That there is a way to use the Datadog Admission Controller with ddtrace. 

I thought I found the appropriate place in the [code where ddtrace figures out how to connect](https://github.com/DataDog/dd-trace-py/blob/fb8dfa2f33fff37d21df9728d8386c0260df9744/ddtrace/internal/agent.py#L65), and it has the correct logic. But I assume this is not actually the code doing this because it does not match the results I get above.

One further point. I was hoping that `DD_TRACE_DEBUG` would help debug how it was trying to connect but it doesn't report anything useful in this area.

### PR DESCRIPTION
**Pull request by brettlangdon**
`ddtrace-run` will try to parse some env variables like `DD_TRACE_AGENT_HOSTNAME`,
`DD_AGENT_HOST`, and `DD_TRACE_AGENT_PORT` and used them to call `tracer.configure()`.

However, the existing default logic when configuring a tracer uses `agent.get_trace_url()`
which knows about these env variables and the appropriate priority order when configuring.

Fixes #3648

### CODE REVIEW

**Comment by brettlangdon** [comment]
@Mergifyio backport 1.1

File: ddtrace/bootstrap/sitecustomize.py [hunk 1]
```diff
@@ -115,10 +106,6 @@ def update_patched_modules():
 
         patch_all(**EXTRA_PATCHED_MODULES)
 
-    dd_env = os.getenv("DD_ENV")
```
Comment (majorgreys) [inline ddtrace/bootstrap/sitecustomize.py:118]: Not sure I follow why this change is here. Where is this tag being set?

**Review by majorgreys — APPROVED** [review]

File: releasenotes/notes/fix-ddtrace-run-url-envs-087ab03d4f25a732.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    tracing: fix issue with ``ddtrace-run`` changing the priority order of tracer host/port/url env variable configuration.
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/fix-ddtrace-run-url-envs-087ab03d4f25a732.yaml:4]:
```suggestion
    tracing: fix issue with ``ddtrace-run`` having the wrong priority order of tracer host/port/url env variable configuration.
```

?

File: ddtrace/bootstrap/sitecustomize.py [hunk 3]
```diff
@@ -98,12 +95,7 @@ def update_patched_modules():
         trace_enabled = True
     else:
         trace_enabled = False
-        opts["enabled"] = False
```
Comment (Kyle-Verhoog) [inline ddtrace/bootstrap/sitecustomize.py:101]: Why this change?

Comment (brettlangdon) [reply ddtrace/bootstrap/sitecustomize.py:101]: I can revert, but it isn't needed. We read `DD_TRACE_ENABLED` for tracer init anyways.

https://github.com/DataDog/dd-trace-py/blob/e27f60181/ddtrace/tracer.py#L187-L187

Comment (brettlangdon) [reply ddtrace/bootstrap/sitecustomize.py:101]: We can revert to keep this PR minimal.

Comment (Kyle-Verhoog) [reply ddtrace/bootstrap/sitecustomize.py:101]: let's follow up with it!

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]
